src/Connector.SqlServer/Connector/SqlClient.cs

```diff
@@ -22,17 +23,62 @@ public string BuildConnectionString(IReadOnlyDictionary<string, object> config)
                 DataSource = (string)config[SqlServerConstants.KeyName.Host],
                 InitialCatalog = (string)config[SqlServerConstants.KeyName.DatabaseName],
                 Pooling = true,
-                MaxPoolSize = 200
             };
 
-            if (config.TryGetValue(SqlServerConstants.KeyName.PortNumber, out var portEntry) && int.TryParse(portEntry.ToString(), out var port))
+            // Configure port
+            {
+                var port = _defaultPort;
+                if (config.TryGetValue(SqlServerConstants.KeyName.PortNumber, out var portEntry) &&
+                    !string.IsNullOrEmpty(portEntry.ToString()) &&
+                    int.TryParse(portEntry.ToString(), out var parsedPort))
+                {
+                    port = parsedPort;
+                }
+
                 connectionStringBuilder.DataSource = $"{connectionStringBuilder.DataSource},{port}";
-            else
-                connectionStringBuilder.DataSource = $"{connectionStringBuilder.DataSource},{_defaultPort}";
+            }
+
+            // Configure connection pool size
+            {
+                var connectionPoolSize = _defaultConnectionPoolSize;
+                if (config.TryGetValue(SqlServerConstants.KeyName.ConnectionPoolSize, out var connectionPoolSizeEntry) &&
+                    !string.IsNullOrEmpty(connectionPoolSizeEntry.ToString()) &&
+                    int.TryParse(connectionPoolSizeEntry.ToString(), out var parsedConnectionPoolSize))
+                {
+                    connectionPoolSize = parsedConnectionPoolSize;
+                }
```

Comment:
Should properly not allow a large connection pool size than the default.
Also add a check for a minimum size as well.

Reply:
Since we're not doing any validation of any of the other values in this method, I think it would be weird to start doing it here. It's already possible to setup an invalid configuration, since the UI allows you to save, even if connection verification fails.